src/Diagnostics.DataProviders/LogAnalyticsClientBase.cs

```diff
@@ -70,11 +71,36 @@ private DataTable ResultAsDataTable(QueryResults results)
             DataTable dataTable = new DataTable("results");
             dataTable.Clear();
 
-            dataTable.Columns.AddRange(results.Tables[0].Columns.Select(s => new DataColumn(s.Name)).ToArray());
+            dataTable.Columns.AddRange(results.Tables[0].Columns.Select(s => new DataColumn(s.Name, StringTypeConverter.StringToType(s.Type))).ToArray());
             var rows = results.Tables[0].Rows.Select(s => dataTable.NewRow().ItemArray = s.ToArray());
-            foreach (var i in rows) { dataTable.Rows.Add(i); }
+            foreach (var i in rows) 
+            {
+                dataTable.Rows.Add(MakeRow(i, dataTable).ToArray());
+            }
 
             return dataTable;
         }
+
+        ArrayList MakeRow(object[] i, DataTable dataTable)
+        {
+            ArrayList variableTypeRow = new ArrayList();
+
+            for (var j = 0; j < i.Length; j++)
+            {
+                if (string.IsNullOrEmpty((string)i[j]))
+                {
+                    variableTypeRow.Add(i[j]);
+                }
+                else if (Type.GetType(dataTable.Columns[j].DataType.FullName) == typeof(DateTime))
+                {
+                    variableTypeRow.Add(DateTime.Parse((string)i[j]).ToUniversalTime());
```

Comment:
dont convert to Universal time automatically. Users might get data in local time.